flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphTranslator.java

```diff
@@ -0,0 +1,95 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.flink.streaming.api.graph;
+
+import org.apache.flink.api.common.Plan;
+import org.apache.flink.api.dag.Pipeline;
+import org.apache.flink.client.FlinkPipelineTranslator;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.optimizer.DataStatistics;
+import org.apache.flink.optimizer.Optimizer;
+import org.apache.flink.optimizer.plan.OptimizedPlan;
+import org.apache.flink.optimizer.plantranslate.JobGraphGenerator;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * {@link FlinkPipelineTranslator} for DataStream API {@link StreamGraph StreamGraphs}.
+ *
+ * <p>Note: this is used through reflection in
+ * {@link org.apache.flink.client.FlinkPipelineTranslationUtil}.
+ */
+@SuppressWarnings("unused")
+public class StreamGraphTranslator implements FlinkPipelineTranslator {
+
+	private static final Logger LOG = LoggerFactory.getLogger(StreamGraphTranslator.class);
+
+	@Override
+	public JobGraph translateToJobGraph(
+			Pipeline pipeline,
+			Configuration optimizerConfiguration,
+			int defaultParallelism) {
+		checkArgument(pipeline instanceof StreamGraph,
+				"Given pipeline is not a DataStream StreamGraph.");
+
+		StreamGraph streamGraph = (StreamGraph) pipeline;
+		return streamGraph.getJobGraph(null);
+	}
+
+	private void setDefaultParallelism(Plan plan, int defaultParallelism) {
+		if (defaultParallelism > 0 && plan.getDefaultParallelism() <= 0) {
+			LOG.debug("Changing plan default parallelism from {} to {}",
+					plan.getDefaultParallelism(),
+					defaultParallelism);
+			plan.setDefaultParallelism(defaultParallelism);
+		}
+
+		LOG.debug("Set parallelism {}, plan default parallelism {}",
+				defaultParallelism,
+				plan.getDefaultParallelism());
+	}
+
+	@Override
+	public String translateToJSONExecutionPlan(Pipeline pipeline) {
+		checkArgument(pipeline instanceof StreamGraph,
+				"Given pipeline is not a DataStream StreamGraph.");
+
+		StreamGraph streamGraph = (StreamGraph) pipeline;
+
+		return streamGraph.getStreamingPlanAsJSON();
+	}
+
+	private JobGraph compilePlan(Plan plan, Configuration optimizerConfiguration) {
```

Comment:
Doesn't used anywhere. Also I don't think StreamGraphTranslator should care about batch `Plan`

Reply:
you're right, I thought I found all of these unused code parts 😓